src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -1873,25 +1873,24 @@ bool DoorLockServer::findUserIndexByCredential(chip::EndpointId endpointId, Cred
     return false;
 }
 
-Status DoorLockServer::createUser(chip::EndpointId endpointId, chip::FabricIndex creatorFabricIdx, chip::NodeId sourceNodeId,
-                                  uint16_t userIndex, const Nullable<chip::CharSpan> & userName,
-                                  const Nullable<uint32_t> & userUniqueId, const Nullable<UserStatusEnum> & userStatus,
-                                  const Nullable<UserTypeEnum> & userType, const Nullable<CredentialRuleEnum> & credentialRule,
-                                  const Nullable<CredentialStruct> & credential)
+StatusIB DoorLockServer::createUser(chip::EndpointId endpointId, chip::FabricIndex creatorFabricIdx, chip::NodeId sourceNodeId,
```

Comment:
We need to make these methods return ClusterStatusCode

And add plumbing to call `CommandHandler::AddClusterSpecificFailure(const ConcreteCommandPath & aCommandPath, ClusterStatus aClusterStatus)` if there is a cluster-specific code.

The real solution is to make CommandHandler and WriteHandler have a AddClusterScecificStatus, which we have not yet done.

I'm actually concerned that the prior code in this cluster did not always properly give the cluster-specific status.

Reply:
It did it in a atrocious static_cast to EmberAfStatus.... But yeah, will replace StatusIB to ClusterStatusCode, it will however be the first time we use that class in the SDK (beyond unit test of course)

Reply:
done